test/e2e/suite/certificaterequests/approval/userinfo.go

```diff
@@ -45,8 +46,15 @@ var _ = framework.CertManagerDescribe("UserInfo CertificateRequests", func() {
 	It("should appropriately create set UserInfo of CertificateRequests, and reject changes", func() {
 		var (
 			adminUsername = "kubernetes-admin"
-			adminGroups   = []string{"system:masters", "system:authenticated"}
 		)
+		// Kubeadm >= 1.29 changed the groups of the admin user from
+		// system:masters to kubeadm:cluster-admins, so instead of hard coding
+		// the group names we try and read them from the client certificate.
+		// https://github.com/kubernetes/kubeadm/issues/2414
+		cert, err := pki.DecodeX509CertificateBytes(f.KubeClientConfig.CertData)
+		Expect(err).NotTo(HaveOccurred())
+		adminGroups := append([]string{"system:authenticated"}, cert.Subject.Organization...)
```

Comment:
The kubernetes config documentation uses  the singular name `client-certificate` ,
so I assumed that this will not be a chain.
 * https://kubernetes.io/docs/reference/config-api/kubeconfig.v1/#AuthInfo

I also considered using the impersonation mechanism to change the groups to something well known here, 
but couldn't see an easy way to impersonate different groups just for this test.

Reply:
I guess all we need is the leaf, so the chain doesn't matter too much either way!